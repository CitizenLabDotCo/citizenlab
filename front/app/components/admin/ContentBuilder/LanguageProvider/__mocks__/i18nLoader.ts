let translations: Record<string, Record<string, string>> = {
  en: { message: 'Hello' },
  'fr-FR': { message: 'Bonjour' },
};

export const i18nImports = new Proxy(
  {},
  {
    get(_, localePath: string) {
      const match = localePath.match(/\/i18n\/(.+)\.ts/);
      if (!match) {
        return undefined;
      }

      const locale = match[1];
      if (locale in translations) {
        return () =>
          Promise.resolve({
            default: translations[locale],
          });
      }

      return undefined;
    },
  }
);

export const setTranslations = (newTranslations: typeof translations) => {
  translations = newTranslations;
};
