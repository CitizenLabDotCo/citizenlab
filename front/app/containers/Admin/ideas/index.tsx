import React, { useState } from 'react';

// module
import { ITab } from 'typings';

// components
import HelmetIntl from 'components/HelmetIntl';
import TabbedResource from 'components/admin/TabbedResource';
import { Outlet as RouterOutlet } from 'react-router-dom';

// i18n
import messages from './messages';
import { WrappedComponentProps } from 'react-intl';
import { injectIntl } from 'utils/cl-intl';

const IdeasPage = ({ intl: { formatMessage } }: WrappedComponentProps) => {
  const [tabs] = useState<ITab[]>([
    {
      label: formatMessage(messages.tabManage),
      name: 'manage',
      url: '/admin/ideas',
    },
  ]);

  const resource = {
    title: formatMessage(messages.inputManagerPageTitle),
    subtitle: formatMessage(messages.inputManagerPageSubtitle),
  };

  return (
    <>
      <TabbedResource resource={resource} tabs={tabs}>
        <HelmetIntl
          title={messages.inputManagerMetaTitle}
          description={messages.inputManagerMetaDescription}
        />
        <div id="e2e-input-manager-container">
          <RouterOutlet />
        </div>
      </TabbedResource>
    </>
  );
};

export default injectIntl(IdeasPage);
