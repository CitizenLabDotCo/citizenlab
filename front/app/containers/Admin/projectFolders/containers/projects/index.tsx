import React from 'react';
// style
import styled from 'styled-components';
// hooks
import useAuthUser from 'hooks/useAuthUser';
// services
import { isAdmin } from 'services/permissions/roles';
// localisation
import { FormattedMessage } from 'utils/cl-intl';
import { withRouter, WithRouterProps } from 'utils/cl-router/withRouter';
import { isNilOrError } from 'utils/helperUtils';
// components
import { HeaderTitle } from 'containers/Admin/projects/all/StyledComponents';
import messages from '../messages';
import ItemsInFolder from './ItemsInFolder';
import ItemsNotInFolder from './ItemsNotInFolder';

const Container = styled.div`
  min-height: 60vh;
`;

const ListsContainer = styled.div``;

const ListHeader = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-start;
  margin-bottom: 25px;

  &:not(:first-child) {
    margin-top: 70px;
  }

  & + & {
    margin-top: 30px;
  }
`;

const StyledHeaderTitle = styled(HeaderTitle)`
  font-weight: bold;
`;

const Spacer = styled.div`
  flex: 1;
`;

const AdminFolderProjectsList = ({
  params: { projectFolderId },
}: WithRouterProps) => {
  const authUser = useAuthUser();

  if (isNilOrError(authUser)) {
    return null;
  }

  const userIsAdmin = authUser && isAdmin({ data: authUser });

  return (
    <Container>
      <ListsContainer>
        <ListHeader>
          <StyledHeaderTitle>
            <FormattedMessage {...messages.projectsAlreadyAdded} />
          </StyledHeaderTitle>
          <Spacer />
        </ListHeader>

        <ItemsInFolder projectFolderId={projectFolderId} />

        {userIsAdmin && (
          <>
            <ListHeader>
              <StyledHeaderTitle>
                <FormattedMessage {...messages.projectsYouCanAdd} />
              </StyledHeaderTitle>
            </ListHeader>

            <ItemsNotInFolder projectFolderId={projectFolderId} />
          </>
        )}
      </ListsContainer>
    </Container>
  );
};

export default withRouter(AdminFolderProjectsList);
