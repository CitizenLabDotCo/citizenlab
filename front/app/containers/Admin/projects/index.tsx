import React from 'react';
import HelmetIntl from 'components/HelmetIntl';
import { Outlet as RouterOutlet } from 'react-router-dom';

// i18n
import messages from './messages';

const AdminProjectsAndFolders = () => (
  <>
    <HelmetIntl
      title={messages.helmetTitle}
      description={messages.helmetDescription}
    />
    <div id="e2e-projects-admin-container">
      <RouterOutlet />
    </div>
  </>
);

export default AdminProjectsAndFolders;
