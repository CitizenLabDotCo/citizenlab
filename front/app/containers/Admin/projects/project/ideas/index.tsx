// Libraries
import React from 'react';
import styled from 'styled-components';
import messages from '../messages';
import { adopt } from 'react-adopt';

// Utils
import { FormattedMessage } from 'utils/cl-intl';

// Resources
import GetPhases, { GetPhasesChildProps } from 'resources/GetPhases';

// Components
import { SectionTitle, SectionDescription } from 'components/admin/Section';
import PostManager, { TFilterMenu } from 'components/admin/PostManager';

// resources
import { isNilOrError } from 'utils/helperUtils';
import { withRouter, WithRouterProps } from 'utils/cl-router/withRouter';
import GetProject, { GetProjectChildProps } from 'resources/GetProject';

const StyledDiv = styled.div`
  margin-bottom: 30px;
`;

interface InputProps {}

interface DataProps {
  phases: GetPhasesChildProps;
  project: GetProjectChildProps;
}

interface Props extends InputProps, DataProps {}

interface State {}

class AdminProjectIdeas extends React.PureComponent<
  Props & WithRouterProps,
  State
> {
  render() {
    const { project, phases } = this.props;
    const defaultTimelineProjectVisibleFilterMenu = 'phases';
    const defaultContinuousProjectVisibleFilterMenu = 'statuses';
    const timelineProjectVisibleFilterMenus: TFilterMenu[] = [
      defaultTimelineProjectVisibleFilterMenu,
      'statuses',
      'topics',
    ];
    const continuousProjectVisibleFilterMenus: TFilterMenu[] = [
      defaultContinuousProjectVisibleFilterMenu,
      'topics',
    ];

    return (
      <>
        <StyledDiv>
          <SectionTitle>
            <FormattedMessage {...messages.titleInputManager} />
          </SectionTitle>
          <SectionDescription>
            <FormattedMessage {...messages.subtitleInputManager} />
          </SectionDescription>
        </StyledDiv>

        {!isNilOrError(project) && (
          <PostManager
            type="ProjectIdeas"
            projectId={project.id}
            phases={phases}
            visibleFilterMenus={
              project.attributes.process_type === 'timeline'
                ? timelineProjectVisibleFilterMenus
                : continuousProjectVisibleFilterMenus
            }
            defaultFilterMenu={
              project.attributes.process_type === 'timeline'
                ? defaultTimelineProjectVisibleFilterMenu
                : defaultContinuousProjectVisibleFilterMenu
            }
          />
        )}
      </>
    );
  }
}

const Data = withRouter(
  adopt<Props>({
    phases: ({ params, render }) => (
      <GetPhases projectId={params.projectId}>{render}</GetPhases>
    ),
    project: ({ params, render }) => (
      <GetProject projectId={params.projectId}>{render}</GetProject>
    ),
  })
);

export default () => (
  <Data>{(dataProps) => <AdminProjectIdeas {...dataProps} />}</Data>
);
