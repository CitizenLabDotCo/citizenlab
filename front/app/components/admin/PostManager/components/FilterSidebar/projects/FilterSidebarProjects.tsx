import React from 'react';

import { Box, Divider, Text } from '@citizenlab/cl2-component-library';

import { IProjectData } from 'api/projects/types';

import Button from 'components/UI/ButtonWithLink';

import { FormattedMessage } from 'utils/cl-intl';

import messages from '../../../messages';
import FilterRadioButton from '../FilterRadioButton';

import FilterSidebarProjectsItem from './FilterSidebarProjectsItem';

interface Props {
  projects?: IProjectData[] | null;
  selectedProject?: string | null;
  onChangeProjectFilter?: (project: string[] | undefined) => void;
}

const FilterSidebarProjects = ({
  projects,
  selectedProject,
  onChangeProjectFilter,
}: Props) => {
  const handleItemClick = (id: string) => () => {
    const projectIds = [id];
    onChangeProjectFilter && onChangeProjectFilter(projectIds);
  };

  const clearFilter = () => {
    onChangeProjectFilter && onChangeProjectFilter(undefined);
  };

  const isActive = (id: string) => {
    return selectedProject === id;
  };

  const name = 'projects';

  return (
    <Box display="flex" flexDirection="column">
      {!(projects && projects.length === 1) && (
        <>
          {/* FilterRadioButton is also used inside FilterSidebarProjectsItem */}
          <FilterRadioButton
            id="all-projects"
            labelContent={<FormattedMessage {...messages.allProjects} />}
            name={name}
            onChange={clearFilter}
            isSelected={!selectedProject}
          />
          <Divider />
        </>
      )}
      <Box display="inline-flex">
        <Button
          buttonStyle="text"
          icon="edit"
          pl="12px"
          linkTo="/admin/projects"
          iconPos="right"
          iconSize="14px"
        >
          <Text m="0px" color="coolGrey600" fontSize="s" textAlign="left">
            <FormattedMessage {...messages.editProjects} />
          </Text>
        </Button>
      </Box>
      {projects &&
        projects.map((project) => (
          <FilterSidebarProjectsItem
            key={project.id}
            project={project}
            active={isActive(project.id)}
            onClick={handleItemClick(project.id)}
            name={name}
          />
        ))}
    </Box>
  );
};

export default FilterSidebarProjects;
