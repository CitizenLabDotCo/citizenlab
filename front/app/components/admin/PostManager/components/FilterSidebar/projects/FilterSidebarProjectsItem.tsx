import React from 'react';

import { useDrop } from 'react-dnd';

import { IProjectData } from 'api/projects/types';

import T from 'components/T';

import FilterRadioButton from '../FilterRadioButton';

interface Props {
  project: IProjectData;
  active: boolean;
  onClick: any;
  name: string;
}

const FilterSidebarProjectsItem = ({
  project,
  active,
  onClick,
  name,
}: Props) => {
  const [{ canDrop, isOver }, drop] = useDrop({
    accept: 'IDEA',
    drop: () => ({
      type: 'project',
      id: project.id,
    }),
    collect: (monitor) => ({
      isOver: monitor.isOver(),
      canDrop: monitor.canDrop(),
    }),
  });

  return (
    <div ref={drop} className="e2e-idea-manager-project-filter-item">
      <FilterRadioButton
        id={project.id}
        name={name}
        isSelected={active || (isOver && canDrop)}
        onChange={onClick}
        labelContent={<T value={project.attributes.title_multiloc} />}
      />
    </div>
  );
};

export default FilterSidebarProjectsItem;
