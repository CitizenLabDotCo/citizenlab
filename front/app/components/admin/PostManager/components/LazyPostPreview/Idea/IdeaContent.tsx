import React, { PureComponent } from 'react';
import { isNilOrError } from 'utils/helperUtils';
import { adopt } from 'react-adopt';
import { get } from 'lodash-es';

// components
import Title from 'components/PostShowComponents/Title';
import PostedBy from 'containers/IdeasShow/PostedBy';
import Body from 'components/PostShowComponents/Body';
import IdeaProposedBudget from 'containers/IdeasShow/IdeaProposedBudget';
import DropdownMap from 'components/PostShowComponents/DropdownMap';
import OfficialFeedback from 'components/PostShowComponents/OfficialFeedback';
import Comments from 'components/PostShowComponents/Comments';
import FileAttachments from 'components/UI/FileAttachments';
import FeedbackSettings from './FeedbackSettings';
import VotePreview from './VotePreview';
import { IconTooltip } from 'cl2-component-library';
import Button from 'components/UI/Button';
import Link from 'utils/cl-router/Link';
import T from 'components/T';
import { Top, Content, Container } from '../PostPreview';

// services
import { deleteIdea } from 'services/ideas';
import { ProcessType } from 'services/projects';

// resources
import GetResourceFiles, {
  GetResourceFilesChildProps,
} from 'resources/GetResourceFiles';
import GetIdea, { GetIdeaChildProps } from 'resources/GetIdea';
import GetIdeaImages, {
  GetIdeaImagesChildProps,
} from 'resources/GetIdeaImages';
import GetLocale, { GetLocaleChildProps } from 'resources/GetLocale';
import GetProject, { GetProjectChildProps } from 'resources/GetProject';
import GetPermission, {
  GetPermissionChildProps,
} from 'resources/GetPermission';

// utils
import { getAddressOrFallbackDMS } from 'utils/map';

// i18n
import injectLocalize, { InjectedLocalized } from 'utils/localize';
import { injectIntl, FormattedMessage } from 'utils/cl-intl';
import { InjectedIntlProps } from 'react-intl';
import messages from '../messages';
import FormattedBudget from 'utils/currency/FormattedBudget';

// style
import styled from 'styled-components';
import { colors, fontSizes } from 'utils/styleUtils';
import { darken } from 'polished';

const StyledTitle = styled(Title)`
  margin-bottom: 20px;
`;

const StyledPostedBy = styled(PostedBy)`
  margin-bottom: 20px;
`;

const Row = styled.div`
  display: flex;
  width: 100%;
`;

const Left = styled.div`
  flex: 5;
  margin-right: 50px;
  height: 100%;
`;

const BelongsToProject = styled.p`
  width: 100%;
  color: ${colors.label};
  font-weight: 300;
  font-size: ${fontSizes.base}px;
  line-height: normal;
  margin-bottom: 10px;
`;

const ProjectLink = styled(Link)`
  color: inherit;
  font-weight: 400;
  font-size: inherit;
  line-height: inherit;
  text-decoration: underline;
  transition: all 100ms ease-out;
  margin-left: 4px;

  &:hover {
    color: ${darken(0.2, colors.label)};
    text-decoration: underline;
  }
`;

const IdeaImage = styled.img`
  width: 100%;
  margin: 0 0 2rem;
  padding: 0;
  border-radius: 8px;
  border: 1px solid ${colors.separation};
`;

const StyledBody = styled(Body)`
  margin-bottom: 20px;
`;

const BodySectionTitle = styled.h2`
  font-size: ${(props) => props.theme.fontSizes.medium}px;
  font-weight: 400;
  line-height: 28px;
`;

const StyledMap = styled(DropdownMap)`
  margin-bottom: 40px;
`;

const StyledOfficialFeedback = styled(OfficialFeedback)`
  margin-top: 70px;
`;

const StyledComments = styled(Comments)`
  margin-top: 30px;
`;

const Right = styled.div`
  flex: 2;
  position: sticky;
  top: 80px;
  align-self: flex-start;
  color: ${colors.adminTextColor};
  font-size: ${fontSizes.base}px;
  line-height: 19px;
`;

const BudgetBox = styled.div`
  margin-top: 25px;
  width: 100%;
  height: 95px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin-bottom: 5px;
  position: relative;
  border-radius: 5px;
  background: ${colors.background};
  border: solid 1px ${colors.adminTextColor};
  font-size: ${fontSizes.large}px;
  font-weight: 500;
`;

const Picks = styled.div`
  margin-top: 15px;
  display: flex;
  font-size: ${fontSizes.base}px;
  align-items: center;
`;

interface State {}

export interface InputProps {
  ideaId: string | null;
  closePreview: () => void;
  handleClickEdit: () => void;
}

interface DataProps {
  idea: GetIdeaChildProps;
  ideaImages: GetIdeaImagesChildProps;
  ideaFiles: GetResourceFilesChildProps;
  locale: GetLocaleChildProps;
  project: GetProjectChildProps;
  postOfficialFeedbackPermission: GetPermissionChildProps;
}

interface Props extends InputProps, DataProps {}

export class IdeaContent extends PureComponent<
  Props & InjectedLocalized & InjectedIntlProps,
  State
> {
  handleClickDelete = (processType: ProcessType) => () => {
    const {
      idea,
      closePreview,
      intl: { formatMessage },
    } = this.props;
    const deleteConfirmationMessage = {
      continuous: messages.deleteInputConfirmation,
      timeline: messages.deleteInputInTimelineConfirmation,
    }[processType];

    if (!isNilOrError(idea)) {
      if (window.confirm(formatMessage(deleteConfirmationMessage))) {
        deleteIdea(idea.id);
        closePreview();
      }
    }
  };

  render() {
    const {
      idea,
      project,
      localize,
      ideaImages,
      ideaFiles,
      locale,
      handleClickEdit,
    } = this.props;

    if (
      !isNilOrError(idea) &&
      !isNilOrError(locale) &&
      !isNilOrError(project)
    ) {
      const ideaId = idea.id;
      const ideaTitle = localize(idea.attributes.title_multiloc);
      const ideaImageLarge =
        !isNilOrError(ideaImages) && ideaImages.length > 0
          ? get(ideaImages[0], 'attributes.versions.large', null)
          : null;
      const ideaGeoPosition = idea.attributes.location_point_geojson || null;
      const ideaAddress = getAddressOrFallbackDMS(
        idea.attributes.location_description,
        idea.attributes.location_point_geojson
      );
      // AuthorId can be null if user has been deleted
      const authorId = idea.relationships.author.data?.id || null;
      const proposedBudget = idea.attributes.proposed_budget;
      const processType = project.attributes.process_type;

      return (
        <Container>
          <Top>
            <Button icon="edit" buttonStyle="text" onClick={handleClickEdit}>
              <FormattedMessage {...messages.edit} />
            </Button>
            <Button
              icon="delete"
              buttonStyle="text"
              onClick={this.handleClickDelete(processType)}
            >
              <FormattedMessage {...messages.delete} />
            </Button>
          </Top>
          <Content>
            {!isNilOrError(project) && (
              <BelongsToProject>
                <FormattedMessage
                  {...messages.postedIn}
                  values={{
                    projectLink: (
                      <ProjectLink
                        className="e2e-project-link"
                        to={`/projects/${project.attributes.slug}`}
                      >
                        <T value={project.attributes.title_multiloc} />
                      </ProjectLink>
                    ),
                  }}
                />
              </BelongsToProject>
            )}

            <StyledTitle postId={ideaId} postType="idea" title={ideaTitle} />
            <StyledPostedBy ideaId={ideaId} authorId={authorId} />
            <Row>
              <Left>
                {ideaImageLarge && (
                  <IdeaImage
                    src={ideaImageLarge}
                    alt=""
                    className="e2e-ideaImage"
                  />
                )}

                {proposedBudget && (
                  <>
                    <BodySectionTitle>
                      <FormattedMessage {...messages.proposedBudgetTitle} />
                    </BodySectionTitle>
                    <IdeaProposedBudget proposedBudget={proposedBudget} />
                    <BodySectionTitle>
                      <FormattedMessage {...messages.bodyTitle} />
                    </BodySectionTitle>
                  </>
                )}

                <StyledBody
                  postId={ideaId}
                  postType="idea"
                  body={localize(idea.attributes.body_multiloc)}
                  locale={locale}
                />

                {!isNilOrError(project) && ideaGeoPosition && ideaAddress && (
                  <StyledMap
                    address={ideaAddress}
                    position={ideaGeoPosition}
                    projectId={project.id}
                  />
                )}

                {ideaFiles && !isNilOrError(ideaFiles) && (
                  <FileAttachments files={ideaFiles} />
                )}

                <StyledOfficialFeedback
                  postId={ideaId}
                  postType="idea"
                  permissionToPost
                />

                <StyledComments postId={ideaId} postType="idea" />
              </Left>
              <Right>
                <VotePreview ideaId={ideaId} />

                {idea.attributes.budget && (
                  <>
                    <BudgetBox>
                      <FormattedBudget value={idea.attributes.budget} />
                      <Picks>
                        <FormattedMessage
                          {...messages.picks}
                          values={{
                            picksNumber: idea.attributes.baskets_count,
                          }}
                        />
                        &nbsp;
                        <IconTooltip
                          content={
                            <FormattedMessage
                              {...messages.pbItemCountTooltip}
                            />
                          }
                        />
                      </Picks>
                    </BudgetBox>
                  </>
                )}

                <FeedbackSettings ideaId={ideaId} />
              </Right>
            </Row>
          </Content>
        </Container>
      );
    }
    return null;
  }
}

const Data = adopt<DataProps, InputProps>({
  locale: <GetLocale />,
  idea: ({ ideaId, render }) => <GetIdea ideaId={ideaId}>{render}</GetIdea>,
  project: ({ idea, render }) => (
    <GetProject projectId={get(idea, 'relationships.project.data.id')}>
      {render}
    </GetProject>
  ),
  ideaFiles: ({ ideaId, render }) => (
    <GetResourceFiles resourceId={ideaId} resourceType="idea">
      {render}
    </GetResourceFiles>
  ),
  ideaImages: ({ ideaId, render }) => (
    <GetIdeaImages ideaId={ideaId}>{render}</GetIdeaImages>
  ),
  postOfficialFeedbackPermission: ({ project, render }) => (
    <GetPermission
      item={!isNilOrError(project) ? project : null}
      action="moderate"
    >
      {render}
    </GetPermission>
  ),
});

const IdeaContentWithHOCs = injectIntl(injectLocalize(IdeaContent));

const WrappedIdeaContent = (inputProps: InputProps) => (
  <Data {...inputProps}>
    {(dataProps) => <IdeaContentWithHOCs {...inputProps} {...dataProps} />}
  </Data>
);

export default WrappedIdeaContent;
