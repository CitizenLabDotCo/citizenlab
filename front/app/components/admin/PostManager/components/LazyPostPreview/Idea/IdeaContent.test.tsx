// @ts-nocheck
import React from 'react';
import { shallow } from 'enzyme';
import { localizeProps } from 'utils/testUtils/localizeProps';
import { getDummyIntlObject } from 'utils/testUtils/mockedIntl';
import { getIdea } from 'services/__mocks__/ideas';
import { getAppConfigurationData } from 'services/__mocks__/appConfiguration';
import { getLocale } from 'services/__mocks__/locale';
import { getProject } from 'services/__mocks__/projects';

jest.mock('containers/IdeasShow/PostedBy', () => 'PostedBy');
jest.mock(
  'containers/IdeasShow/IdeaProposedBudget',
  () => 'IdeaProposedBudget'
);
jest.mock('components/PostShowComponents/Title', () => 'Title');
jest.mock('components/PostShowComponents/Body', () => 'Body');
jest.mock('components/PostShowComponents/DropdownMap', () => 'DropdownMap');
jest.mock(
  'components/PostShowComponents/OfficialFeedback',
  () => 'OfficialFeedback'
);
jest.mock('components/PostShowComponents/Comments', () => 'Comments');
jest.mock('components/UI/FileAttachments', () => 'FileAttachments');
jest.mock('components/UI/Button', () => 'Button');
jest.mock('cl2-component-library', () => ({
  IconTooltip: 'IconTooltip',
  Icon: 'Icon',
}));
jest.mock('../PostPreview', () => ({
  Top: 'Top',
  Content: 'Content',
  Container: 'Container',
}));
jest.mock('./FeedbackSettings', () => 'FeedbackSettings');
jest.mock('./VotePreview', () => 'VotePreview');
jest.mock('utils/cl-router/Link', () => 'Link');
jest.mock('components/T', () => 'T');
jest.mock('services/ideas');
jest.mock('resources/GetResourceFiles', () => 'GetResourceFiles');
jest.mock('resources/GetIdea', () => 'GetIdea');
jest.mock('resources/GetIdeaImages', () => 'GetIdeaImages');
jest.mock('resources/GetAppConfiguration', () => 'GetAppConfiguration');
jest.mock('resources/GetProject', () => 'GetProject');
jest.mock('resources/GetPermission', () => 'GetPermission');
jest.mock('resources/GetLocale', () => 'GetLocale');
jest.mock('utils/cl-intl');
jest.mock('utils/styleUtils', () => ({
  colors: {
    label: '#000',
    separation: '#000',
    adminTextColor: '#000',
  },
  fontSizes: {
    base: 16,
  },
  media: {
    smallerThanMinTablet: (x) => x,
  },
}));

jest.mock('modules', () => ({ streamsToReset: [] }));

import { IdeaContent } from './IdeaContent';

describe('<IdeaContent />', () => {
  let closePreview: jest.Mock;
  let handleClickEdit: jest.Mock;

  beforeEach(() => {
    closePreview = jest.fn();
    handleClickEdit = jest.fn();
  });

  it('renders correctly when ideaId is not defined', () => {
    const intl = getDummyIntlObject();
    const project = getProject();
    const wrapper = shallow(
      <IdeaContent
        ideaId={null}
        closePreview={closePreview}
        handleClickEdit={handleClickEdit}
        project={project}
        idea={null}
        ideaImages={null}
        ideaFiles={null}
        tenant={null}
        locale={null}
        postOfficialFeedbackPermission={undefined}
        intl={intl}
        {...localizeProps}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly when an idea is provided', () => {
    const ideaId = 'myIdeasiD';
    const idea = getIdea(ideaId);
    const locale = getLocale();
    const tenant = getAppConfigurationData();
    const project = getProject();
    const intl = getDummyIntlObject();
    const wrapper = shallow(
      <IdeaContent
        ideaId={ideaId}
        closePreview={closePreview}
        handleClickEdit={handleClickEdit}
        idea={idea}
        tenant={tenant}
        locale={locale}
        project={project}
        ideaImages={null}
        ideaFiles={null}
        postOfficialFeedbackPermission={undefined}
        intl={intl}
        {...localizeProps}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
