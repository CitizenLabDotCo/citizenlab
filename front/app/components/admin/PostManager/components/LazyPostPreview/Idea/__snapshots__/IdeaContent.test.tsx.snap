// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IdeaContent /> renders correctly when an idea is provided 1`] = `
<Container>
  <Top>
    <Button
      buttonStyle="text"
      icon="edit"
      onClick={[MockFunction]}
    >
      <FormattedMessage
        defaultMessage="Edit"
        id="app.components.admin.PostManager.PostPreview.edit"
      />
    </Button>
    <Button
      buttonStyle="text"
      icon="delete"
      onClick={[Function]}
    >
      <FormattedMessage
        defaultMessage="Delete"
        id="app.components.admin.PostManager.PostPreview.delete"
      />
    </Button>
  </Top>
  <Content>
    <IdeaContent__BelongsToProject>
      <FormattedMessage
        defaultMessage="Posted in {projectLink}"
        id="app.containers.IdeasShow.postedIn"
        values={
          Object {
            "projectLink": <IdeaContent__ProjectLink
              className="e2e-project-link"
              to="/projects/undefined_slug"
            >
              <T
                value={
                  Object {
                    "en": "undefined project Title",
                  }
                }
              />
            </IdeaContent__ProjectLink>,
          }
        }
      />
    </IdeaContent__BelongsToProject>
    <IdeaContent__StyledTitle
      postId="myIdeasiD"
      postType="idea"
      title="Doloribus quam molestiae ut."
    />
    <IdeaContent__StyledPostedBy
      authorId="8ea3a61c-e1bd-4756-a54d-c092553abccd"
      ideaId="myIdeasiD"
    />
    <IdeaContent__Row>
      <IdeaContent__Left>
        <IdeaContent__BodySectionTitle>
          <FormattedMessage
            defaultMessage="Proposed budget"
            id="app.containers.IdeasShow.proposedBudgetTitle"
          />
        </IdeaContent__BodySectionTitle>
        <IdeaProposedBudget
          proposedBudget={45}
        />
        <IdeaContent__BodySectionTitle>
          <FormattedMessage
            defaultMessage="Description"
            id="app.containers.IdeasShow.bodyTitle"
          />
        </IdeaContent__BodySectionTitle>
        <IdeaContent__StyledBody
          body="<p>Ea ea aut. Quasi corrupti iste. Delectus eum voluptates.</p><p>Est harum voluptates. Et fugit enim. Suscipit inventore ullam.</p><p>Eos mollitia omnis. Laudantium porro perferendis. Minus in aut.</p>"
          locale="en"
          postId="myIdeasiD"
          postType="idea"
        />
        <IdeaContent__StyledMap
          address="50° 52′ 8.38577″ N 4° 25′ 7.43365″ E"
          position={
            Object {
              "coordinates": Array [
                4.418731568531502,
                50.86899604801978,
              ],
              "type": "Point",
            }
          }
        />
        <IdeaContent__StyledOfficialFeedback
          permissionToPost={true}
          postId="myIdeasiD"
          postType="idea"
        />
        <IdeaContent__StyledComments
          postId="myIdeasiD"
          postType="idea"
        />
      </IdeaContent__Left>
      <IdeaContent__Right>
        <VotePreview
          ideaId="myIdeasiD"
        />
        <IdeaContent__BudgetBox>
          <FormattedBudget
            value={60}
          />
          <IdeaContent__Picks>
            <FormattedMessage
              defaultMessage="Picks: {picksNumber}"
              id="app.components.admin.PostManager.PostPreview.picks"
              values={
                Object {
                  "picksNumber": 0,
                }
              }
            />
             
            <IconTooltip
              content={
                <FormattedMessage
                  defaultMessage="The number of times this has been included in other participants' participatory budgets"
                  id="app.components.admin.PostManager.PostPreview.pbItemCountTooltip"
                />
              }
            />
          </IdeaContent__Picks>
        </IdeaContent__BudgetBox>
        <FeedbackSettings
          ideaId="myIdeasiD"
        />
      </IdeaContent__Right>
    </IdeaContent__Row>
  </Content>
</Container>
`;

exports[`<IdeaContent /> renders correctly when ideaId is not defined 1`] = `""`;
